pyiron_gui/project/project.py

```diff
@@ -0,0 +1,51 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from pyiron_base import Project as BaseProject
+
+from pyiron_gui.project.project_browser import ProjectBrowser
+
+__author__ = "Niklas Siemer"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Niklas Siemer"
+__email__ = "siemer@mpie.de"
+__status__ = "development"
+__date__ = "Feb 02, 2021"
+
+
+def activate_gui(project_instance):
+    """Activate GUI elements for a provided pyiron Project instance
+
+        Args:
+            project_instance: Instantiated pyiron Project
+        Returns:
+            GUIProject: Instantiated pyiron Project with added GUI functionality.
+    """
+
+    class GUIProject(project_instance.__class__):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self._project_browser = None
+
+        @property
+        def browser(self):
+            """Provides a browser to inspect the data system of the project."""
+            if self._project_browser is None:
+                self._project_browser = ProjectBrowser(project=self,
```

Comment:
Nit: args all fit on one 100-character line? I dont' have it open in my IDE just the web browser, but it looks like they'll fit